great_expectations/analytics/events.py

```diff
@@ -1,13 +1,115 @@
 from __future__ import annotations
 
+from dataclasses import dataclass
 from typing import ClassVar, List
 
-from great_expectations.analytics.actions import DATA_CONTEXT_INITIALIZED
+from great_expectations.analytics.actions import (
+    DATA_CONTEXT_INITIALIZED,
+    EXPECTATION_SUITE_CREATED,
+    EXPECTATION_SUITE_DELETED,
+    EXPECTATION_SUITE_EXPECTATION_CREATED,
+    EXPECTATION_SUITE_EXPECTATION_DELETED,
+    EXPECTATION_SUITE_EXPECTATION_UPDATED,
+)
 from great_expectations.analytics.base_event import Action, Event
+from great_expectations.compatibility.typing_extensions import override
 
 
 class DataContextInitializedEvent(Event):
+    """
+    Emitted when a DataContext is initialized.
+      - AbstractDataContext.__init__
+    """
+
     _allowed_actions: ClassVar[List[Action]] = [DATA_CONTEXT_INITIALIZED]
 
     def __init__(self):
         super().__init__(action=DATA_CONTEXT_INITIALIZED)
+
+
+@dataclass
+class _ExpectationSuiteExpectationEvent(Event):
+    expectation_id: str | None = None
+    expectation_suite_id: str | None = None
+
+    @override
+    def _properties(self) -> dict:
+        return {
+            "expectation_id": self.expectation_id,
+            "expectation_suite_id": self.expectation_suite_id,
+        }
```

Comment:
Feels like this is always just the properties on the object? We could make this a lot simpler by iterating through `self.__dict__` or something?

Reply:
i would use the builtin dataclass function, unless there's a strong reason not to